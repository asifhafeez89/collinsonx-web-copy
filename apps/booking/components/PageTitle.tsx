import Head from 'next/head';
import { SITE_NAME } from 'config';

export interface PageTitleProps {
  title: string;
}
const PageTitle = ({ title }: PageTitleProps) => {
  return (
    <Head>
      <title>{title ? title + ' - ' + SITE_NAME : SITE_NAME}</title>
    </Head>
  );
};

export default PageTitle;
