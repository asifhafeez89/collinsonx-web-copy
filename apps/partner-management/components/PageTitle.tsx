import useExperience from 'hooks/experience';
import Head from 'next/head';
import { useMemo } from 'react';

import { useSessionContext } from 'supertokens-auth-react/recipe/session';
import { SUPER_USER } from 'config';

export interface PageTitleProps {
  section?: string;
  title: string;
  customFormat?: boolean;
}
const PageTitle = ({
  title,
  section,
  customFormat = false,
}: PageTitleProps) => {
  const session = useSessionContext();
  const isLoggedIn = !session.loading && session.doesSessionExist;

  const isSuperUser = useMemo(() => {
    if (isLoggedIn) {
      return (session.accessTokenPayload ?? {}).userType === SUPER_USER;
    }
  }, [session, isLoggedIn]);

  const { experience } = useExperience();

  const partnerName = isSuperUser ? 'Collinson' : experience?.loungeName ?? '';

  let result = title;
  if (!customFormat && isLoggedIn) {
    result = `${title}${section ? ' | ' + section : ''}: ${partnerName}`;
  }
  return (
    <Head>
      <title>{result}</title>
    </Head>
  );
};

export default PageTitle;
