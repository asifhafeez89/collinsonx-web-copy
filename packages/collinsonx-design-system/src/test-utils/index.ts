import userEvent from '@testing-library/user-event';

export * from '@testing-library/react';
export { userEvent };
export { default as render, Provider } from './render';
