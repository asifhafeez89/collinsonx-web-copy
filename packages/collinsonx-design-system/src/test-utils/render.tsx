import { RenderOptions, render } from '@testing-library/react';
import { MantineProvider } from '@mantine/core';
import { ReactElement, ReactNode } from 'react';

export const Provider = ({ children }: { children: ReactNode }) => (
  <MantineProvider>{children}</MantineProvider>
);

type Render = (
  ui: ReactElement,
  options?: Omit<RenderOptions, 'wrapper'>
) => ReturnType<typeof render>;

const customRender: Render = (ui, options) =>
  render(ui, {
    wrapper: Provider as unknown as RenderOptions['wrapper'],
    ...options,
  });

export default customRender;
