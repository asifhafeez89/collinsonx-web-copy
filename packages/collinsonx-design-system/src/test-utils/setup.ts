import userEvent from '@testing-library/user-event';
import render from './render';

export default function (jsx: Parameters<typeof render>[0]) {
  return {
    user: userEvent.setup(),
    ...render(jsx),
  };
}
