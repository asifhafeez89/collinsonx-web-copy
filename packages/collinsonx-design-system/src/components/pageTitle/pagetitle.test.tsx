import renderer from 'react-test-renderer';
import { Provider } from 'test-utils';
import PageTitle from '.';

describe('<PageTitle />', () => {
  it('renders PageTitle', () => {
    const tree = renderer
      .create(
        <Provider>
          <PageTitle title="test" onClickBack={() => {}} />
        </Provider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
