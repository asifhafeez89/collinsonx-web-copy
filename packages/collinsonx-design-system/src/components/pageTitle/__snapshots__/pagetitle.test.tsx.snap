// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PageTitle /> renders PageTitle 1`] = `
<header
  className="mantine-1fvi2kf"
>
  <a
    href="https://bbc.co.uk"
    onClick={[Function]}
    onMouseEnter={[Function]}
    onTouchStart={[Function]}
  >
    <div
      className="mantine-1e5u164"
    >
      <test-file-stub
        color="#000000"
      />
    </div>
  </a>
  <h1
    className="mantine-Text-root mantine-Title-root mantine-57g7lp"
  >
    test
  </h1>
</header>
`;
